src/core/sock.c

```diff
@@ -1088,3 +1161,24 @@ void nn_sock_stat_increment (struct nn_sock *self, int name, int64_t increment)
             break;
     }
 }
+
+int nn_sock_hold (struct nn_sock *self)
+{
+    switch (self->state) {
+    case NN_SOCK_STATE_ACTIVE:
+    case NN_SOCK_STATE_INIT:
+        self->holds++;
+        return 0;
+    case NN_SOCK_STATE_ZOMBIE:
+        return -ETERM;
+    }
```

Comment:
So I'm adding the explicit cases along with default.  For default I put it to EBADF as well.  I'm not a huge fan of littering the code with explicit checks / assertions; in fact I think a lot of the fragility we've seen in nanomsg in the past has been the result of too 'stringent' checks on things like errnos.

In this case, if its not a state that I explicitly covered already, then I want to return EBADF.